src/analysis/polarity_search.ml

```diff
@@ -138,3 +151,57 @@ let execute_query query env dirs =
       acc
   in
   List.fold_left dirs ~init:(direct None []) ~f:recurse
+
+let execute_query_as_type_search
+    ?(limit = 100)
+    ~config
+    ~local_defs
+    ~comments
+    ~pos
+    ~env
+    ~query
+    ~modules () =
+  let direct dir acc =
+    Env.fold_values (fun _ path desc acc ->
+        let d  = desc.Types.val_type in
+        match match_query env query d with
+        | Some cost ->
+          let path = Printtyp.rewrite_double_underscore_paths env path in
+          let name = Format.asprintf "%a" Printtyp.path path in
+          let doc =
+            Locate.get_doc
+              ~config
+              ~env
+              ~local_defs
+              ~comments
+              ~pos
+              (`User_input name)
+            |> Type_search.doc_to_option
```

Comment:
I am wondering about the cost of fetching the doc of all the returned items. It's not a trivial operation. Maybe we should provide an option to the command so that clients can opt-out from that. (I put the comment here but of course it is also valable for the new type-search results

Reply:
What do you think about the approach proposed here https://github.com/ocaml/merlin/pull/1828/commits/a8688aaacb24b15b4cec612915d4a018d918c33c